apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/RpcRetryConsumer.cs

```diff
@@ -72,7 +72,10 @@ internal class RpcRetryConsumer<T> : IConsumer<T>
             _timerFactory = GaxPreconditions.CheckNotNull(timerFactory, nameof(timerFactory));
 
             // Start the background timer that will retry failed requests.
-            _timer = _timerFactory(FlushBuffer, _options);
+            if (options.RetryType != RetryType.None)
+            {
```

Comment:
is there value in asserting the interval here where its closer to what would cause the issue?  Or are you forced to always go thru the retryoption validation?